Code/GraphMol/molopstest.cpp

```diff
@@ -6978,6 +6978,222 @@ void testGithub1703() {
   BOOST_LOG(rdInfoLog) << "Finished" << std::endl;
 }
 
+void testGithub1614() {
+  BOOST_LOG(rdInfoLog) << "-----------------------\n Testing github issue "
+                          "1614: AssignStereochemistry incorrectly removing "
+                          "CIS/TRANS bond stereo"
+                       << std::endl;
+#if 1
+  {
+    RWMol m;
+    m.addAtom(new Atom(9), true, true);
+    m.addAtom(new Atom(6), true, true);
+    m.addAtom(new Atom(6), true, true);
+    m.addAtom(new Atom(17), true, true);
+    m.addBond(0, 1, Bond::SINGLE);
+    m.addBond(2, 3, Bond::SINGLE);
+    m.addBond(1, 2, Bond::DOUBLE);
+    m.getBondBetweenAtoms(1, 2)->setStereoAtoms(0, 3);
+    m.getBondBetweenAtoms(1, 2)->setStereo(Bond::STEREOTRANS);
+    m.updatePropertyCache();
+
+    {
+      RWMol nm(m);
+      bool force = true, cleanIt = true;
+      MolOps::setDoubleBondNeighborDirections(nm);
```

Comment:
It looks like we'll be requiring that setDoubleBondNeighborDirections be called after manually setting Bond::setStereo(STEREOCIS/STEREOTRANS)? That seems fine as the function name is quite explicitly named, but I'm curious about why this design is necessary?

@greglandrum, did you try folding MolOps::setDoubleBondNeighborDirections into the preamble of MolOps::assignStereochemistry? Would this be a bad idea?

Reply:
It's not so much that it's a bad idea, but I (think) I've tried to keep the bond directions unset whenever possible. So double bond stereo is stored on the double bonds themselves, and not set redundantly on their neighboring bonds unless client code requests it.

Reply:
That seems reasonable to never set bond directions until they're needed (really only the SMILES writer?). Are they a prerequisite for MolOps::assignStereochemistry now though?

Reply:
hmm, looks like it is still necessary for assignStereochemistry.

You definitely don't need it to get correct SMILES:
```
In [13]: m = Chem.MolFromSmiles('CC=CC')

In [14]: m.GetBondWithIdx(1).SetStereoAtoms(0,3)

In [15]: m.GetBondWithIdx(1).SetStereo(Chem.BondStereo.STEREOCIS)

In [16]: Chem.MolToSmiles(m,True)
Out[16]: 'C/C=C\\C'
```

but to have AssignStereochemistry() work it's necessary:
```
In [24]: m = Chem.MolFromSmiles('CC=CC')

In [25]: m.GetBondWithIdx(1).SetStereoAtoms(0,3)

In [26]: m.GetBondWithIdx(1).SetStereo(Chem.BondStereo.STEREOCIS)

In [27]: Chem.AssignStereochemistry(m,cleanIt=True,force=True)

In [28]: m.GetBondWithIdx(1).GetStereo()
Out[28]: rdkit.Chem.rdchem.BondStereo.STEREONONE
```
That feels like a bug (though not necessarily one to hold up this merge). What's your take?

Reply:
Not necessary to hold up this merge. I'll play with it in the context of expanding EnumerateStereoisomers for pseudo-stereo and let you know how it goes there.